sstables/sstable_directory.cc

```diff
@@ -343,9 +344,22 @@ future<> sstable_directory::system_keyspace_components_lister::commit() {
     return make_ready_future<>();
 }
 
-future<> sstable_directory::system_keyspace_components_lister::garbage_collect() {
-    // FIXME -- implement
-    co_return;
+future<> sstable_directory::system_keyspace_components_lister::garbage_collect(storage& st) {
+    return do_with(std::set<generation_type>(), [this, &st] (auto& gens_to_remove) {
```

Comment:
can we coroutinize this part? so we don't need the `do_with()` .

Reply:
It was stepping on https://github.com/scylladb/seastar/issues/1625 in v1 so I gave up and just coded it as then-chains

Reply:
ah, very true. i now recall the discussion we had the other day.